### PR TITLE
Remove hard dependency on libpthread

### PR DESCRIPTION
**Pull request by ueno**
This implements the suggestion in: https://bugzilla.redhat.com/show_bug.cgi?id=985495#c0

### CODE REVIEW

**Comment by ueno** [comment]
@fweimer would you take a look at this when you have time?

File: common/compat.c [hunk 1]
```diff
@@ -169,6 +169,14 @@ getprogname (void)
 #include <fcntl.h>
 #include <unistd.h>
 
+#ifdef PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+static pthread_mutex_t mutex_initializer = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+void
+p11_mutex_init (p11_mutex_t *mutex)
+{
+	memcpy (mutex, &mutex_initializer, sizeof (pthread_mutex_t));
```
Comment (fweimer) [inline common/compat.c]: I think this is technically undefined behavior. Copying of mutex objects is not supported.

Comment (ueno) [reply common/compat.c]: Thank you.  The initializer is only applicable to statically allocated mutexes, right?  In p11-kit there are a few places where dynamically allocated mutexes are used; how could we initialize them?

Comment (ueno) [reply common/compat.c]: After going through the uses of mutexes, I realized that only `p11_library_mutex` (and possibly `p11_virtual_mutex`) needs to be recursive (see 563606efe17cbf3b84679f5e54f60b8d68ba9015). I have changed all other initialization to use the default attribute.

**Comment by ueno** [comment]
Thank you for the review. I have replaced the use of `pthread_atfork()` with `__register_atfork()`, which glibc provides,

**Pull request merged**